components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRManagementService.java

```diff
@@ -147,7 +147,10 @@ private RegistrationResponseProfile createOAuthApplication(RegistrationRequestPr
                     createdServiceProvider = appMgtService.getServiceProvider(applicationName,
                                                                               profile.getTenantDomain());
                 } else {
-                    createdServiceProvider = existingServiceProvider;
+                    String errorMessage = "Service Provider Application: " + applicationName +
```

Comment:
Better to change the error message to "Service Provider with name: applicationName already registered".